components/contribution-flow/StepDetails.js

```diff
@@ -126,8 +127,31 @@ const StepDetails = ({ onChange, data, collective, tier, showPlatformTip, router
                 currencyDisplay="full"
                 prependProps={{ color: 'black.500' }}
                 required
+                onKeyDown={event => {
+                  setPressedKey(event.key);
+                }}
```

Comment:
Seems a bit hacky to rely on the state for this, could we rather look at the event directly in `onChange`?

Reply:
Actually no, the event in `onChange` don't have the `key` property available.

Reply:
The `event.nativeEvent.data` seems to have the number that was typed. And since we know that this logic is ultimately [going to move](https://github.com/opencollective/opencollective/issues/6381#issuecomment-1411100288) to `StyledInputAmount`, it seems like the easiest option to implement it there directly.

The only thing it won't support is the logic implemented by `pressedKey === 'Backspace' || pressedKey === 'Delete'` but I don't really understand what we're solving there?

Reply:
> The event.nativeEvent.data seems to have the number that was typed.

I didn't know about this, let me check if we can update the logic to use it instead.

> The only thing it won't support is the logic implemented by pressedKey === 'Backspace' || pressedKey === 'Delete' but I don't really understand what we're solving there?

I've added these since the input value can be changed with these keys too, let me check if there is a case in which there can be an increment/decrement of 1 by using `Backspace` if not we can remove this.

Reply:
> The only thing it won't support is the logic implemented by pressedKey === 'Backspace' || pressedKey === 'Delete' but I don't really understand what we're solving there.
 
If I type `20.51` and then hit `Backspace` the value decreases by `0.5` to `20.01`. This would be a rare case though.


https://user-images.githubusercontent.com/46647141/218380778-6e7741ca-3588-4f81-b6b0-a54581190ae5.mov